test/library/cert_utils.go

```diff
@@ -121,8 +122,8 @@ func VerifyCertificate(secret *v1.Secret, commonName string) (bool, error) {
 	}
 
 	// check certificate subject CN
-	if cert.Subject.CommonName != commonName {
-		return false, fmt.Errorf("incorrect subject CN: %v found in issued certificate", cert.Subject.CommonName)
+	if strings.ToLower(cert.Subject.CommonName) != strings.ToLower(commonName) {
```

Comment:
@geliu2016 Did you come across an edge case where this if condition was getting violated without the addition of `strings.ToLower`, just curious!

Reply:
Yes, 
print err log in this func as below, you may find the original random_str is: GQZqCtH, but get from func: "cert, err := GetX509Certificate(secret)", the GQZqCtH is changed to "gqzqcth", 

<*errors.errorString | 0xc00043d6b0>: 
      incorrect subject CN: gqzqcth.geliu41512.qe.gcp.devcluster.openshift.com found in issued certificate, different from GQZqCtH.geliu41512.qe.gcp.devcluster.openshift.com
      {
          s: "incorrect subject CN: gqzqcth.geliu41512.qe.gcp.devcluster.openshift.com found in issued certificate, different from GQZqCtH.geliu41512.qe.gcp.devcluster.openshift.com",
      }

Reply:
Then we should instead make sure that the random string generator function only returns lowercase characters. As DNS names are generally preferred lowercase.

Reply:
> Note that while upper and lower case letters are allowed in domain
names, no significance is attached to the case.  That is, two names with
the same spelling but different case are to be treated as if identical.

Excerpt from https://www.ietf.org/rfc/rfc1035.txt